src/explore-education-statistics-frontend/src/components/__tests__/Feedback.test.tsx

```diff
@@ -0,0 +1,133 @@
+import Feedback from '@frontend/components/Feedback';
+import { render, screen, waitFor } from '@testing-library/react';
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import _feedbackService from '@frontend/services/feedbackService';
+import { FeedbackRequest } from '@common/services/types/feedback';
+
+jest.mock('@frontend/services/feedbackService');
+
+const feedbackService = jest.mocked(_feedbackService);
+
+describe('Feedback', () => {
+  test('renders initial state correctly', () => {
+    render(<Feedback />);
+    expect(screen.getByText('Is this page useful?')).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Yes this page is useful' }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'No this page is not useful' }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Report a problem with this page' }),
+    ).toBeInTheDocument();
+  });
+
+  test('renders correctly and submits when "Yes" is selected', async () => {
+    render(<Feedback />);
+    await userEvent.click(
+      screen.getByRole('button', { name: 'Yes this page is useful' }),
+    );
+
+    await waitFor(() => {
+      expect(feedbackService.sendFeedback).toHaveBeenCalledWith(
+        expect.objectContaining<Omit<FeedbackRequest, 'userAgent'>>({
+          url: '/',
+          response: 'Useful',
+        }),
+      );
+    });
+
+    expect(screen.getByText('Thank you for your feedback')).toBeInTheDocument();
+  });
+
+  test('renders correctly when "No" is selected', async () => {
+    render(<Feedback />);
+    await userEvent.click(
+      screen.getByRole('button', { name: 'No this page is not useful' }),
+    );
+
+    expect(screen.getByLabelText('What were you doing?')).toBeInTheDocument();
+    expect(
+      screen.getByLabelText('What were you hoping to achieve?'),
+    ).toBeInTheDocument();
```

Comment:
We should check that these aren't visible before the button is pressed as well to check that a change actually occurred

Same applies for equivalent 'Report a problem' test case below as well